pragma-node/src/infra/repositories/onchain_repository.rs

```diff
@@ -79,10 +80,38 @@ impl From<SpotEntryWithAggregatedPrice> for OnchainEntry {
     }
 }
 
-fn get_aggregation_query(aggregation_mode: AggregationMode) -> Result<&'static str, InfraError> {
+impl From<&SpotEntryWithAggregatedPrice> for OnchainEntry {
+    fn from(entry: &SpotEntryWithAggregatedPrice) -> Self {
+        OnchainEntry {
+            publisher: entry.spot_entry.publisher.to_owned(),
```

Comment:
why `to_owned` instead of `clone`?

Reply:
We dont need those data and to_owned just transfert the ownership to another variable meanwhile clone is creating a new instance and make a deep copy.
It should be better memory and performance wise but maybe im wrong

Reply:
`to_owned` actually clones:
- https://doc.rust-lang.org/std/borrow/trait.ToOwned.html
- https://doc.rust-lang.org/1.80.0/src/alloc/borrow.rs.html#83-85

Reply:
oh ok mb, will change then

Reply:
done